internal/mocks/nmockapi/domain_client.go

```diff
@@ -17,6 +17,12 @@ import (
 // environments.
 type DomainClient struct {
 	baseClient[*ngrok.ReservedDomain]
+
+	// Error injection fields for testing
+	createError error
```

Comment:
I wanted to be able to simulate errors to check false conditions. And the clientset above is because the cloudendpoitn controller uses a clientset unlike the domain controller that just takes the domains client. @jonstacks  let me know if this follows the pattern for mocking the ngrok client that we are aiming for. I saw some mock gen but didn't see it really being used